lua/clear-action/signs.lua

```diff
@@ -5,19 +5,29 @@ local utils = require("clear-action.utils")
 
 local clear_extmark = function() vim.api.nvim_buf_clear_namespace(0, config.ns, 0, -1) end
 
-local function on_result(results, context)
+local function on_result(results_all, line, bufnr)
+  local result_count = 0
+  local result_actions_flat = {}
+  for client_id, results in pairs(results_all) do
```

Comment:
As I suggested in the issue maybe we could append the client id to each action, that would allow us to get the action provider's name in the `label_fmt` function for example

Reply:
yep agree. so this var (`flat`) will gain a client_id field in it to identify the client and user can define a `label_fmt` which can make a `vim.lsp.get_client_by_id` call to render that out.